test/semant/Tests/ScopingSpec.hs

```diff
@@ -0,0 +1,263 @@
+{-# LANGUAGE QuasiQuotes #-}
```

Comment:
This test suite tests the scoping, and includes plenty of example programs

It also demonstrates that now we are able to test that certain kinds of programs fail in a specific way, so we don't get false positives (e.g., we are trying to test that the compiler catches type errors, but it actually fails due to a scope error).